src/libstd/keyword_docs.rs

```diff
@@ -1030,9 +1030,85 @@ mod self_upper_keyword {}
 //
 /// A place that is valid for the duration of a program.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// A `static` item is similar to a [`const`] item in that it lives for the
+/// entire duration of the program and need to have its type explicited, with a
+/// `static` lifetime, outliving any other lifetime. Added to that, `static`
+/// items represent a precise memory location.
 ///
-/// [not yet complete]: https://github.com/rust-lang/rust/issues/34601
+/// Static items do not call [`drop`] at the end of the program.
+///
+/// There are two types of `static` items: those declared in association with
+/// the [`mut`] keyword and those without.
+///
+/// # Simple `static`s
+///
+/// Non-[`mut`] `static` items that contain a type that is not interior mutable
+/// may be placed in read-only memory. All access to a `static` item are
+/// considered safe but some restrictions apply. See the [Reference] for more
+/// information.
+///
+/// ```rust
+/// static FOO: [i32; 5] = [1, 2, 3, 4, 5];
+///
+/// let r1 = &FOO as *const _;
+/// let r2 = &FOO as *const _;
+/// // With a strictly read-only static, references will have the same adress
+/// assert_eq!(r1, r2);
+/// ```
+///
+/// # Mutable `static`s
+///
+/// If a `static` item is declared with the [`mut`] keyword, then it is allowed
+/// to be modified by the program. To make concurrency bugs hard to run into,
+/// all access to a `static mut` require an [`unsafe`] block. Care should be
+/// taken to ensure access (both read and write) are thread-safe.
+///
+/// Despite their unsafety, mutable `static`s are very useful: they can be used
+/// to represent global state shared by the whole program or be used in
+/// [`extern`] blocks to bind to variables from C libraries.
```

Comment:
Again, sentence from the reference, but I wouldn't sound so optimistic about statics :D They are pretty rarely needed; most Rust programmers won't need them at all. So I would tone this down somehow, e.g. "very useful" -> "necessary in many contexts".

Reply:
I may have been biased since I'm doing FFI for fun on the side, you're right a more neutral tone is better suited to docs.

Reply:
And I'm probably biased since I basically never worked with FFI in Rust :D